src/attributes.jl

```diff
@@ -1133,13 +1133,41 @@ struct ListOfConstraintAttributesSet{F,S} <: AbstractModelAttribute end
 """
     ConstraintName()
 
-A constraint attribute for a string identifying the constraint. It is *valid*
-for constraints variables to have the same name; however, constraints with
-duplicate names cannot be looked up using [`get`](@ref) regardless of if they
-have the same `F`-in-`S` type. It has a default value of `""` if not set.
+A constraint attribute for a string identifying the constraint.
+
+It is *valid* for constraints variables to have the same name; however,
+constraints with duplicate names cannot be looked up using [`get`](@ref),
+regardless of whether they have the same `F`-in-`S` type.
+
+`ConstraintName` has a default value of `""` if not set.
+
+## Notes
+
+You should _not_ implement `ConstraintName` for `SingleVariable` constraints.
 """
 struct ConstraintName <: AbstractConstraintAttribute end
 
+"""
+    SingleVariableConstraintNameError()
+
+An error to be thrown when the user tries to set `ConstraintName` on a
+`SingleVariable` constraint.
+"""
+function SingleVariableConstraintNameError()
+    return UnsupportedAttribute(
+        ConstraintName(),
+        "`ConstraintName`s are not supported for `SingleVariable` constraints.",
+    )
+end
+
+function supports(
+    ::ModelLike,
+    ::ConstraintName,
+    ::Type{ConstraintIndex{SingleVariable,S}},
+) where {S}
+    return throw(SingleVariableConstraintNameError())
```

Comment:
This is going to cause ambiguities because it is both broad in the first argument and wide in the second and last one. We should redirect `supports` to a `supports_fallback` to avoid it.

Reply:
> This is going to cause ambiguities

Not really? Only if someone defines `set(::Optimizer, ::ConstraintName, ::ConstraintIndex)` which is invalid because it doesn't handle the SingleVariable case.

The documentation also gives two functions to implement that would resolve any ambiguities.

Reply:
If you are an MOI layer, you would want to pass the call along and trust the inner optimizer to throw the error.
That would cause ambiguity

Reply:
Then you can just implement the functions listed in the documentation to resolve it?

Reply:
Yes but having to implement this function for all MOI layers is a lot more work than implementing one fallback. That's also consistent with what we already do with `set_fallback`